### DISCUSSION TITLE
How to add custom logic to handle cancel for a SIP invite

### DISCUSSION THREAD

**Discussion by krambojun** [emiago/sipgo#131] [Q&A]
Given that there has been this refactoring https://github.com/emiago/sipgo/commit/806740005bbb0a8681ccc8721a204cf8be1f18da, how to know a sip cancel has been sent so that custom logic can be called?

I tried checking this `dtx.setState(sip.DialogStateEnded)` based on the temp fix added and also leveraged the onCancel method but none of those seem to work

**Comment by emiago** [comment]
hi @krambojun as server, you will have error ErrTransactionCancel, but I am not sure your use case, and where exactly you need this. I believe it gets propagated, although I think it could be dedicated error as well.
This is on `tx.Err()` after  `tx.Done()`
